common/djangoapps/third_party_auth/saml.py

```diff
@@ -310,14 +310,16 @@ def get_user_details(self, attributes):
             sys_msg = err.response.json() if err.response else "Not available"
             log_msg_template = (
                 'Unable to retrieve user details with username {username} from SAPSuccessFactors for company ' +
-                'ID {company} with url "{url}".  Error message: {err_msg}.  System message: {sys_msg}.'
+                'ID {company} with url "{url}".  Error message: {err_msg}.  System message: {sys_msg}.  ' +
+                'Headers: {headers}'
             )
             log_msg = log_msg_template.format(
                 username=username,
                 company=self.odata_company_id,
                 url=odata_api_url,
                 err_msg=err.message,
-                sys_msg=sys_msg
+                sys_msg=sys_msg,
+                headers=err.response.headers
```

Comment:
@mattdrayer I think you need to guard against the exception not having a `response` attribute here like you are doing above. Seeing some errors on stage related this:
```
Nov  6 11:40:36 ip-10-3-11-100 [service_variant=lms][django.request][env:stage-edx-edxapp] ERROR [ip-10-3-11-100  28714] [base.py:256] - Internal Server Error: /auth/complete/tpa-saml/
Traceback (most recent call last):
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/decorators.py", line 145, in inner
    return func(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/newrelic/hooks/framework_django.py", line 527, in wrapper
    return wrapped(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/views/decorators/cache.py", line 57, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/social_django/utils.py", line 50, in wrapper
    return func(request, backend, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/social_django/views.py", line 32, in complete
    redirect_name=REDIRECT_FIELD_NAME, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/social_core/actions.py", line 41, in do_complete
    user = backend.complete(user=user, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/social_core/backends/base.py", line 40, in complete
    return self.auth_complete(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/social_core/backends/saml.py", line 311, in auth_complete
    return self.strategy.authenticate(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/social_django/strategy.py", line 115, in authenticate
    return authenticate(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/contrib/auth/__init__.py", line 74, in authenticate
    user = backend.authenticate(**credentials)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/social_core/backends/base.py", line 80, in authenticate
    return self.pipeline(pipeline, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/social_core/backends/base.py", line 83, in pipeline
    out = self.run_pipeline(pipeline, pipeline_index, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/social_core/backends/base.py", line 108, in run_pipeline
    result = func(*args, **out) or {}
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/social_core/pipeline/social_auth.py", line 5, in social_details
    return {'details': dict(backend.get_user_details(response), **details)}
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/social_core/backends/saml.py", line 274, in get_user_details
    return idp.get_user_details(response['attributes'])
  File "/edx/app/edxapp/edx-platform/common/djangoapps/third_party_auth/saml.py", line 322, in get_user_details
    headers=err.response.headers
AttributeError: 'NoneType' object has no attribute 'headers'
```

Reply:
Yep I'm seeing the errors as well -- I'll submit a PR today that fixes the issue and also adds a bit more logging.